File: 1/python/sol.py
import numpy as np

data = []

with open("../input.txt", "r") as f:
    for line in f:
        data.append(int(line))

data_i = np.array(data)


def part1(data):

    full = np.array([2020]*len(data))
    full = full - data

    for x in full:
        if x in data:
            return x,(2020-x)


def part2(data):
    data = set(data)
    for x in data:
        for y in data:
            if 2020 - x - y in data:
                return x, y, 2020-x-y

if __name__=="__main__":
    print("part 1:", np.prod(part1(data_i)))
    print("part 2:", np.prod(part2(data_i)))

File: 1/haskell/sol.hs
import System.IO
import Control.Monad

readInt :: String -> Int
readInt = read

main = do
  contents <- readFile "../input.txt"
  let nums = map readInt . words $ contents
  let part1 = product ([[x,y]| x <- nums, y  <- nums, x + y == 2020] !! 0)
  let part2 = product ([[x,y,z]| x <- nums, y  <- nums, z <- nums, x + y + z == 2020] !! 0)
  print $ "Part 1: " ++ (show part1)
  print $ "Part 2: " ++ (show part2)
